lineman/app/services/lmo_url_service.coffee

```diff
@@ -1,17 +1,21 @@
 angular.module('loomioApp').factory 'LmoUrlService', ->
   new class LmoUrlService
 
-    stub: (name) ->
-      name.replace(/[^a-z0-9\-_]+/gi, '-').replace(/-+/g, '-').toLowerCase()
+    group: (g, params) ->
+      "/g/#{g.key}/#{@stub(g.fullName())}#{@queryStringFor(params)}"
+
+    discussion: (d, params = {}) ->
+      "/d/#{d.key}/#{@stub(d.title)}#{@queryStringFor(params)}"
 
-    group: (g) ->
-      "/g/#{g.key}/#{@stub(g.fullName())}"
+    proposal: (p, params = {}) ->
+      "/m/#{p.key}/#{@stub(p.name)}#{@queryStringFor(params)}"
```

Comment:
We're just going to link to the thread (ie: /d) soon eh? Is there any reason in your mind why we're hanging onto /m at this stage?